profiler/src/main/java/com/splunk/opentelemetry/profiler/snapshot/SnapshotVolumePropagator.java

```diff
@@ -40,22 +39,24 @@ public Collection<String> fields() {
   public <C> void inject(Context context, C carrier, TextMapSetter<C> setter) {}
 
   /**
-   * We're attempting to infer the start of a trace by inspecting the provided {@link Context}, and
-   * if so make a decision whether to select the trace for snapshotting. We will not have a trace ID
-   * at this time so we'll communicate the snapshotting decision to later instrumentation steps by
-   * placing an entry in OpenTelemetry's {@link io.opentelemetry.api.baggage.Baggage}. <br>
-   * <br>
-   * If we're somewhere downstream of the trace root we leave the {@link Context} unchanged.
+   * Make a decision whether a trace should be selected for snapshotting if --and only if -- that
+   * decision has not yet been made by an upstream service. Ideally a snapshotting decision will
+   * have been made only at trace root, however there are many scenarios (e.g. RUM, or language
+   * agents without snapshot support) where a trace will begin without a decision. In those
+   * instances we would like to snapshot as much of the trace as possible.
+   *
+   * <p>Not seen here in the introduction of {@link TraceIdBasedSnapshotSelector} which will
```

Comment:
Idk whether this is relevant but one side effect of using `TraceIdBasedSnapshotSelector` is that it might produce a surprising result with sampling. Say that you are using the same thread id strategy for sampling traces. If you select 10% traces and use the same strategy to select 10% for snapshotting then you will still end up with 100% of traces that were sampled also be selected for snapshotting.